ouroboros-consensus/test-consensus/Test/Dynamic/General.hs

```diff
@@ -125,10 +126,12 @@ runTestNetwork ::
      )
   => (CoreNodeId -> ProtocolInfo blk)
   -> TestConfig
+  -> Maybe (NodeConfig (BlockProtocol blk) -> SlotNo -> BlockNo -> ChainHash blk -> blk)
```

Comment:
I'd like a type synonym for this, e.g., `EBBProducer blk` or `EbbProducer blk`.

Reply:
Fixed, thanks. I went with a bespoke variant of `Maybe`, so the `Nothing` would be more legible.

(I still intend to repackage these arguments in a record type, so I might undo that then. But these PR seems to have enough moving parts as-is.)